include/miopen/config.h.in

```diff
@@ -78,11 +78,65 @@
 
 // clang-format off
 #cmakedefine MIOPEN_DEFAULT_FIND_MODE @MIOPEN_DEFAULT_FIND_MODE@
+
 // clang-format on
+#cmakedefine01 MIOPEN_OFFLINE_COMPILER_PATHS_V2
+#if MIOPEN_OFFLINE_COMPILER_PATHS_V2
+// Include rocm-core header for get ROCm Install Path Method
+#include <rocm-core/rocm_getpath.h>
+
+// Include header for PATH_MAX
+#if defined(__linux__)
+#include <linux/limits.h>
+#endif
+#if defined(_WIN32)
+#include <windows.h>
+#ifndef PATH_MAX
+#define PATH_MAX MAX_PATH
+#endif
+#endif
+
+// Max Buffer Len to hold the Path
+#define PATH_MAX_LEN (PATH_MAX + 1)
+
+// Flags to help identify/differentiate Definitions for each MIOPEN Compiler Paths
+typedef enum
+{
+    MIOPEN_AMDGCN_Flag          = 0,
+    MIOPEN_OC_COMPILER_Flag     = 1,
+    MIOPEN_HIP_COMPILER_Flag    = 2,
+    MIOPEN_OFFLOAD_BUNDLER_Flag = 3,
+    MIOPEN_CMP_MAX_Flag         = 4 // Should be last Flag definition
+} MIOPEN_CMP_Flags_t;
+
+// Global variable prototype of List to hold the Compiler Flag Paths
+// The definition of this variable expected to be available in C/C++
+// code of the Target Library
+extern char MIOPEN_CMP_FLAGS[MIOPEN_CMP_MAX_Flag][PATH_MAX_LEN];
```

Comment:
Use `fs::path` from `miopen\filesystem.hpp` instead. Linux paths will not work on Windows. `fs::path` will handle separators (and the rest) correctly.